extensions/oidc-client-filter/deployment/src/main/java/io/quarkus/oidc/client/filter/deployment/OidcClientFilterBuildStep.java

```diff
@@ -24,15 +35,88 @@ public class OidcClientFilterBuildStep {
     @BuildStep
     void registerProvider(BuildProducer<AdditionalBeanBuildItem> additionalBeans,
             BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            NamedOidcClientFilterBuildItem namedOidcClientFilterBuildItem,
             BuildProducer<ResteasyJaxrsProviderBuildItem> jaxrsProviders,
-            BuildProducer<RestClientAnnotationProviderBuildItem> restAnnotationProvider) {
+            BuildProducer<RestClientAdditionalProviderBuildItem> restAnnotationProvider) {
+
         additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(OidcClientRequestFilter.class));
         reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, OidcClientRequestFilter.class));
+        final Set<String> namedFilterClientClasses = namedOidcClientFilterBuildItem.namedFilterClientClasses;
+
+        // register default request filter provider against the rest of the clients (client != namedFilterClientClasses)
         if (config.registerFilter) {
-            jaxrsProviders.produce(new ResteasyJaxrsProviderBuildItem(OidcClientRequestFilter.class.getName()));
+            if (namedFilterClientClasses.isEmpty()) {
+                // register default request filter as global rest client provider
+                jaxrsProviders.produce(new ResteasyJaxrsProviderBuildItem(OidcClientRequestFilter.class.getName()));
+            } else {
+                // register all clients without @OidcClientFilter("clientName")
+                restAnnotationProvider
+                        .produce(new RestClientAdditionalProviderBuildItem(OidcClientRequestFilter.class.getName(),
+                                new Predicate<ClassInfo>() {
+                                    @Override
+                                    public boolean test(ClassInfo restClientClassInfo) {
```

Comment:
It does not really matter much, but if you could rename `test` to something more related to the context then it could be easier to read

Reply:
That's `java.util.function.Predicate` and I could only do that with wrapper class, but I can document it better in constructors and `RestClientAdditionalProviderBuildItem` member property `matcher` etc.